In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df=pd.read_csv('sales_add.csv')

In [3]:
df

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [4]:
from scipy import stats as stat


In [5]:
df.info

<bound method DataFrame.info of        Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Regio

In [6]:
df.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [7]:
df.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [8]:
df.columns

Index(['Month', 'Region', 'Manager', 'Sales_before_digital_add(in $)',
       'Sales_After_digital_add(in $)'],
      dtype='object')

# 1
The company wishes to clarify whether there is any increase in sales after
stepping into digital marketing.



Ho: There is increase in sales after digital marketing

H1: There is no increase in sales after digital marketing

In [9]:
from scipy import stats as stat
stat.ttest_rel(df['Sales_After_digital_add(in $)'], df['Sales_before_digital_add(in $)'])

Ttest_relResult(statistic=12.09070525287017, pvalue=6.336667004575778e-11)

# 
Here we got the value for t statistic as 12.09 and p value is near to 0.

Since p value < 0.05, we reject H0. ie; We can conclude that there is no increase in the sales after the digital marketing

# 2 
The company needs to check whether there is any dependency between the
features “Region” and “Manager”.



Ho:  There is dependancy between Region and Manager

H1:  There is no dependancy between Region and  Manager

# 

In [10]:
df['Region'].value_counts() 


Region - A    10
Region - B     7
Region - C     5
Name: Region, dtype: int64

In [11]:
df['Manager'].value_counts() 

Manager - A    9
Manager - B    7
Manager - C    6
Name: Manager, dtype: int64

In [12]:
observations=pd.crosstab(df.Region,df.Manager)
observations

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [13]:
import scipy.stats as stats
stats.chi2_contingency(observations)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

In [14]:
#we get the Chi square value as 3.0506, p value = 0.5494, degrees of freedom = 4.
#here the p value is greater than .005, that is we accept the null hypothysis (failed to reject).
#That means there is dependeny between Region and manager.

In [15]:
import scipy
Table=scipy.stats.chi2.ppf(1-.05, df=4) 
Table

9.487729036781154

In [16]:
# the calculated chi square value is less than the table value, so we fail to reject the null hypothesis